src/CommunityToolkit.Maui.MediaElement/PlatformView/MediaManager.android.cs

```diff
@@ -0,0 +1,255 @@
+				//Com.Google.Android.Exoplayer2.Util.Util.InferContentType()
+				//var httpDataSourceFactory = new DefaultHttpDataSource.Factory();
+				//var mediaSource = new DashMediaSource.Factory(httpDataSourceFactory)
+				//.CreateMediaSource(MediaItem.FromUri(uri));
+
+				//player.SetMediaSource(mediaSource);
+
+				player.SetMediaItem(MediaItem.FromUri(uri));
+				player.Prepare();
+
+				hasSetSource = true;
+			}
+		}
+		else if (mediaElement.Source is FileMediaSource)
+		{
+			string filePath = (mediaElement.Source as FileMediaSource)!.File!;
+			if (!string.IsNullOrWhiteSpace(filePath))
+			{
+				player.SetMediaItem(MediaItem.FromUri(filePath));
+				player.Prepare();
+
+				hasSetSource = true;
+			}
+		}
+
+		if (hasSetSource && mediaElement.AutoPlay)
+		{
+			player.Play();
+		}
+	}
+
+	protected virtual partial void PlatformUpdateSpeed()
+	{
+		if (mediaElement is null || player is null)
+		{
+			return;
+		}
+
+		if (mediaElement.Speed > 0)
+		{
+			player.SetPlaybackSpeed((float)mediaElement.Speed);
+			player.Play();
+		}
+		else
+		{
+			player.Pause();
+		}
+	}
+
+	protected virtual partial void PlatformUpdateShowsPlaybackControls()
+	{
+		if (mediaElement is null || playerView is null)
+		{
+			return;
+		}
+
+		playerView.UseController = mediaElement.ShowsPlaybackControls;
+	}
+	protected virtual partial void PlatformUpdatePosition()
+	{
+		if (mediaElement is null || player is null)
+		{
+			return;
+		}
+
+		if (Math.Abs(player.CurrentPosition - mediaElement.Position.TotalMilliseconds) > 1000)
+		{
+			player.SeekTo((long)mediaElement.Position.TotalMilliseconds);
+		}
+	}
+
+	protected virtual partial void PlatformUpdateStatus()
+	{
+		if (mediaElement is null || player is null)
+		{
+			return;
+		}
+
+		var videoStatus = MediaElementState.Closed;
+
+		switch ((PlaybackStateCode)player.PlaybackState)
+		{
+			case PlaybackStateCode.Playing:
+				videoStatus = MediaElementState.Playing;
+				break;
+
+			case PlaybackStateCode.Paused:
+				videoStatus = MediaElementState.Paused;
+				break;
+
+			case PlaybackStateCode.Stopped:
+				videoStatus = MediaElementState.Stopped;
+				break;
+		}
+
+		mediaElement.CurrentState = videoStatus;
+
+		mediaElement.Position = TimeSpan.FromMilliseconds(player.CurrentPosition);
+	}
+
+	protected virtual partial void PlatformUpdateVolume()
+	{
+		if (mediaElement is null || player is null)
+		{
+			return;
+		}
+
+		player.Volume = (float)mediaElement.Volume;
+	}
+
+
+	protected virtual partial void PlatformUpdateIsLooping()
+	{
+		if (mediaElement is null || player is null)
+		{
+			return;
+		}
+
+		player.RepeatMode = mediaElement.IsLooping ? IPlayer.RepeatModeOne : IPlayer.RepeatModeOff;
+	}
+
+	#region IPlayer.IListener implementation method stubs
+	public void OnPlayerStateChanged(bool playWhenReady, int playbackState)
+	{
+		if (player is null)
+		{
+			return;
+		}
+
+		if (playbackState == IPlayer.StateReady)
+		{
+			mediaElement.Duration = TimeSpan.FromMilliseconds(player.Duration);
```

Comment:
Might be causing exceptions on specific types of streams. I am still investigating, but this is an error I received when trying to play an URL on Android: **System.OverflowException:** 'TimeSpan overflowed because the duration is too long.'. It could be possibly related to what people are discussing here: https://stackoverflow.com/questions/35298125/exoplayer-getduration